docs/DeveloperGuide.md

```diff
@@ -1,25 +1,290 @@
+
+### Edit Function<br/>
+
+#### Class Diagram <br/>
+
+![](images/ClassDiagram_EditCommand.png)<br/>
+
+### Delete Function
+
+#### System Architecture of the Delete Function<br/>
+
+![](images/DeleteCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the delete command.<br/>
+* UI: The User Interface of the app<br/>
+* DeleteCommand: The main logic command of the delete function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the Delete Function<br/>
+
+![](images/DeleteCommand_umlDiagram.png)
+
+1. `DeleteCommand` gets the input from the User and gets the relevant timetable from `Timetable`
+1. `DeleteCommand` then proceeds to delete the event in the timetable based on the index in the user input
+1. `DeleteCommand` passes the day, deleted event and size of timetable to the `UI`
+1. `UI` updates the user of the successful deletion of the event
+
+#### Sequence Design of Delete Function<br/>
+![](images/DeleteCommand_SequenceDiagram.png)<br/>
+
+### Clear Function
+
+#### System Architecture of the Clear Function<br/>
+
+![](images/ClearCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the clear command.<br/>
+* UI: The User Interface of the app<br/>
+* ClearCommand: The main logic command of the clear function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the Clear Function<br/>
+
+![](images/ClearCommand_umlDiagram.png)
+
+1. `ClearCommand` gets the input from the User and gets the relevant timetable from `Timetable`
+1. `ClearCommand` then proceeds to clear all events from the timetable
+1. `ClearCommand` passes the day and cleared lesson to the `UI`
+1. `UI` updates the user of the successful clearing of events
+
+#### Sequence Design of Clear Function<br/>
+![](images/ClearCommand_SequenceDiagram.png)<br/>
+
+### List Function
+
+#### System Architecture of the List Function<br/>
+
+![](images/ListCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the list command.<br/>
+* UI: The User Interface of the app<br/>
+* ListCommand: The main logic command of the list function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the List Function<br/>
+
+![](images/ListCommand_umlDiagram.png)
+
+1. `ListCommand` gets the input from the User and gets the relevant user's timetable(s) from `Timetable`
+1. `ListCommand` then passes the user's timetable(s) to the `UI`
+1. `UI` prints all the classes in the timetable
+
+#### Sequence Design of List Function<br/>
+![](images/ListCommand_SequenceDiagram.png)<br/>
```

Comment:
There should be an activation bar under the Ui class when printList is called. After calling the printList method, arrows should be used to indicate returning from Ui to ListCommand and also from execute() to the ListCommand class itself